diff --git a/bindings/odin/clay-odin/clay.odin b/bindings/odin/clay-odin/clay.odin
index 9332f8b..f92b2b2 100644
--- a/bindings/odin/clay-odin/clay.odin
+++ b/bindings/odin/clay-odin/clay.odin
@@ -4,7 +4,7 @@ import "core:c"
 
 when ODIN_OS == .Windows {
 	foreign import Clay "windows/clay.lib"
-} else when ODIN_OS == .Linux {
+} else when ODIN_OS == .Linux || ODIN_OS == .OpenBSD {
 	foreign import Clay "linux/clay.a"
 } else when ODIN_OS == .Darwin {
 	when ODIN_ARCH == .arm64 {
